src/game/AI/ScriptDevAI/scripts/kalimdor/darkshore.cpp

```diff
@@ -1007,15 +1003,15 @@ bool GOUse_go_furbolg_food(Player* player, GameObject* go)
     // Search for a nearby Blackwood Warrior and make it attack
     if (Creature* warrior = GetClosestCreatureWithEntry(go, NPC_BLACKWOOD_WARRIOR, 20.0f))
     {
-        if (warrior->IsInCombat() || !warrior->IsAlive() || !player->IsEnemy(warrior))
+        if (!warrior->IsInCombat() && warrior->IsAlive() && player->IsEnemy(warrior))
```

Comment:
My *guess* is that this is what caused the crash mentioned in the issue, since I couldn't reproduce it.
My speculation is that if the closest creature with that entry was dead&despawned that either it has no initialized AI() or AttackStart() fails at some point because the creature doesn't exist. Pure speculation tho.

Reply:
That's a fair assumption and would it prove to be true, I would then be the culprit here. It seems that at some point I changed my mind about the logic when adding this safety check but that I did not changed the following code accordingly: asking a creature to attack when it is possibly dead or not an enemy of its target makes absolutely no sense.

Reply:
I suggest using CanAttack instead of IsEnemy, because the former is more than the latter.

Reply:
Done. The reason why IsEnemy was chosen was because the quest script temporarily turns them into a friendly faction, so IsEnemy() is less computationally intensive and achieves the desired effect

Reply:
I understand, but isenemy has a hidden caveat you prolly dont know. Factions arent symmetrical, I can be your enemy, but you can be my friend. CanAttack checks both in the correct manner. This was actually an issue with brutallus last year, because he was neutral for players, but players were hostile to him and some spells werent functioning properly until I adjusted it. All of these functions come directly from client and B basically uses CanAttack everywhere always.